### PR TITLE
Cadc 11901a Rework

### PR DESCRIPTION
**Pull request by yeunga**
- based on Brian's review comments
- refactored SessionAction to improve performance
- added session ID to desktop-app
- fixed DeleteAction which exposed a bug as a result of adding session ID to desktop-app

### CODE REVIEW

File: deployment/k8s-config/kustomize/base/skaha/config/launch-desktop-app.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ spec:
   template:
     metadata:
       labels:
+        canfar-net-sessionID: "${skaha.sessionid}"
```
Comment (brianmajor) [inline deployment/k8s-config/kustomize/base/skaha/config/launch-desktop-app.yaml:16]: For symmetry, this label should also be added to the label section above.  (and in the overlay version of this file)

File: skaha/src/main/java/org/opencadc/skaha/session/DeleteAction.java [hunk 2]
```diff
@@ -105,24 +105,33 @@ public void doAction() throws Exception {
...
+                        // want to ignore them as we pick the session to be deleted.
+                        for (String line : lines) {
+                            String[] parts = line.split("\\s+");
+                            if (!TYPE_DESKTOP_APP.equals(parts[0])) {
+                                String sessionUserId = parts[1];
+                                if (!userID.equals(sessionUserId)) {
+                                    throw new AccessControlException("forbidden");
+                                }   
+    
+                                String type = parts[0];
```
Comment (brianmajor) [inline skaha/src/main/java/org/opencadc/skaha/session/DeleteAction.java]: The type could be extracted (this line) before it's used (parts[0]) on line 116.

**Review by brianmajor — CHANGES_REQUESTED** [review]
Looking good, just a few minor comments

File: deployment/k8s-config/kustomize/base/skaha/config/launch-desktop-app.yaml (hunk 1 above)
Comment (yeunga) [reply deployment/k8s-config/kustomize/base/skaha/config/launch-desktop-app.yaml:16]: Good eye. Added.

File: skaha/src/main/java/org/opencadc/skaha/session/DeleteAction.java (hunk 2 above)
Comment (yeunga) [reply skaha/src/main/java/org/opencadc/skaha/session/DeleteAction.java]: Updated.

**Review by brianmajor — APPROVED** [review]

**Pull request merged**